### DISCUSSION TITLE
[solved] in -gravity center,... | One word red one yellow one green ...

### DISCUSSION THREAD

**Discussion by htohneb** [ImageMagick/ImageMagick#6121] [Help] [answered]
### ImageMagick version

7.1.0-62 Q16-HDRI aarch64 20885

### Operating system, version and so on

mac os x 13.2.1 (compiler gcc 4.2)

### Description

Hello everyone,

i want that the words had different colors.

convert -bordercolor white -border 5x5 -background white -fill blue -font Arial -size 800x1100 \
          -gravity center caption:'This text is resized to best fill the space given.' \
          caption_filled_sm.0.gif

That the text: 'This text is resized to best fill the space given.'

The first word: This = red
The second word: text = yellow
The third word: if = green
.
.
.
how i can do that?

Thank you for your time

**Comment by fmw42** [comment]
You cannot do that with caption:  You would have to use -annotate 3 times in succession in the command line, once for each colored word.  Or Use PANGO:

See https://imagemagick.org/Usage/text/

**Comment by htohneb** [comment]
Ok, thank you [fmw42](https://github.com/fmw42), that was fast!

Pango is installed, but don't run.

I test it with -annotate.

Thank's

**Reply by fmw42** [reply]
What do you mean that Pango does not run.  What do you get from 

magick -version

Does it include Pango or Cairo?  If not, then it is not installed properly. You also need to recompile IM as Pango is a delegate.

**Reply by htohneb** [reply]
The Version: ImageMagick 7.1.0-62 Q16-HDRI aarch64 20885

**Comment by htohneb** [comment]
I want to install it from https://formulae.brew.sh/formula/pango#default

The Terminal messages:

ted@Touch Desktop % brew install pango
==> Downloading https://formulae.brew.sh/api/formula.jws.json
######################################################################## 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
######################################################################## 100.0%
Warning: pango 1.50.13 is already installed and up-to-date.
To reinstall 1.50.13, run:
  brew reinstall pango
ted@Touch Desktop % brew install pango
Warning: pango 1.50.13 is already installed and up-to-date.
To reinstall 1.50.13, run:
  brew reinstall pango

Hmm, it seems the installation is finish

I test it

**Reply by fmw42** [reply]
Sorry, I do not use Homebrew.  I suspect you need to uninstall Imagemagick and uninstall Pango. Then reinstall them together in the same brew command line.

**Reply by htohneb** [reply]
What do you use on mac? If you install additional software?

Thank's for your reply. 

now, it's late for me, I'll look tomorrow, or friday. 

Thank's a lot! I'll reply you,...

**Comment by fmw42** [comment]
I install all my delegates with MacPorts.  Then I install Imagemagick from source.

**Comment by htohneb** [comment]
now I test:
magick -size 400x400 xc:white \
          -fill black -annotate 300x300+20+50 "black" \
          -fill green -annotate 300x300+35+50 "green" \
          -fill orange -annotate 300x300+50+50 "orange" \
          -fill yellow -annotate 300x300+65+50 "yelllow" \
          -fill blue -annotate 300x300+80+50 "Blue" \
          annotated_labels.1.jpg

This text is weird, I got it 1 to 1 from: https://imagemagick.org/Usage/text/#annotate

now I can't get any further.

_What I intend to do:_

**_A line as output, which contains different colors for the guards.
This line is to be installed with gravity center in the middle of an image of size X by Y._**

What i must do do?

Thank you for your time

**Reply by fmw42** [reply]
You must either make it one long single line or put new line / at the end of each line as follows;

```
magick -size 400x400 xc:white \
-fill black -annotate 300x300+20+50 "black" \
-fill green -annotate 300x300+35+50 "green" \
-fill orange -annotate 300x300+50+50 "orange" \
-fill yellow -annotate 300x300+65+50 "yelllow" \
-fill blue -annotate 300x300+80+50 "Blue" \
annotated_labels.1.jpg

```

![annotated_labels 1](https://user-images.githubusercontent.com/13458218/222769917-0298aee2-f8b7-49a4-8d84-bf0f8e0ff73f.jpg)

**Reply by htohneb** [reply]
Thank you fmw42, 

I change to: magick -size 400x400 xc:white -fill black -annotate 0 "black" -fill green -annotate 0 "green" -fill orange -annotate 0 "orange" -fill yellow -annotate 0 "yellow" -fill blue -annotate 0 "Blue" annotated_labels.7.jpg

Thats starts at Position 0x0 correct, and start to write all words at 0x0. 

What I have to do that I'll want to **fill** the picture, with **-graphity center**, with the words, one word behind the next word, with a normal space between them?

Thank you for reading and your time

**Reply by fmw42** [reply]
Is this what you want?  You need to use +Y of 0 and +X must have two negative, one at 0 and two positive relative to center (of 0).

```
magick -size 400x400 xc:white -gravity center \
-fill black -annotate 300x300-30+0 "black" \
-fill green -annotate 300x300-15+0 "green" \
-fill orange -annotate 300x300+0+0 "orange" \
-fill yellow -annotate 300x300+15+0 "yelllow" \
-fill blue -annotate 300x300+30+0 "Blue" \
annotated_labels.2.jpg
```

![annotated_labels 2](https://user-images.githubusercontent.com/13458218/222790735-ab685ca6-9e94-4f52-8c79-7310b75198cd.jpg)

**Reply by htohneb** [reply]
Thank you fmw42,

The image here is what I'll want.

The left and right, and the hight, is in the middst.

If I'll can use 2 space lines, that is perfekt!

Thank you for your reading and your time

<img width="729" alt="Bildschirm­foto 2023-03-03 um 19 31 56" src="https://user-images.githubusercontent.com/69136734/222799793-5885d613-b5d2-47e5-9b87-16f850a5bc63.png">

**Comment by snibgo** [comment, marked as answer]
You could do this in ImageMagick using "`pango:`". See https://imagemagick.org/Usage/text/#pango

However, ImageMagick is not a desktop publishing system or similar. For many purposes, HTML provides all the facilities we need. For rasterizing HTML pages, I suggest https://wkhtmltopdf.org/index.html . I expect that program can be declared as a delegate to ImageMagick, if desired.

**Comment by fmw42** [comment]
To do

yellow word, green word, blue word, red word,

You will need to use Pango.  Annotate can do it, but you have to put the words next to each other and not on different lines. So the +Y value is the same, but the +X value needs to change to align each word by itself. This is not easy, but can be done esp with fixed width fonts

**Comment by htohneb** [comment]
Thank you, [snibgo](https://github.com/snibgo) and [fmw42](https://github.com/fmw42), 

it goes forward!

I know, that I'll youse pango, for what I want to do!

Perfekt

Thank you for reading and your time!
